iree/turbine/kernel/wave/runtime/runtime.cpp

```diff
@@ -65,22 +68,45 @@ int launch(const KernelLaunchInfo &info, const Int64Vector &tensors, const Int64
             cache[info.kernelHash] = std::make_tuple(module, function);
     }
 
+    ScalarKind scalarKind;
+    size_t scalarSize;
+
+    nb::handle first_ele = scalarArgs.size() == 0 ? nb::float_(0.f) : scalarArgs[0];
+    scalarKind = nb::isinstance<nb::int_>(first_ele) ? ScalarKind::I32 : ScalarKind::F32;
+    scalarSize = sizeof(float);  // both int and float dtypes are 4 bytes
+
     // Since we always pass our dynamic dims as index type, iree converts them to i64
     // and then splits them to two i32s, i64 = hi | lo where
     // lo = trunc(i64) and hi = trunc(i64 >> 32).
-    size_t kernArgSize = tensors.size() * sizeof(uint64_t) + 2 * dynamicDims.size() * sizeof(uint32_t);
+    size_t kernArgSize = tensors.size() * sizeof(uint64_t) + 2 * dynamicDims.size() * sizeof(uint32_t) + scalarArgs.size() * scalarSize;
+    std::vector<uint8_t> scalarBytes(scalarArgs.size() * scalarSize);
+
+    uint32_t* dst = reinterpret_cast<uint32_t*>(scalarBytes.data());
+
+    if (scalarKind == ScalarKind::F32) {
+        for (auto&& arg : scalarArgs) {
+            float val = nb::cast<float>(arg);
+            std::memcpy(dst++, &val, sizeof(float));
+        }
+    } else {
+        for (auto&& arg : scalarArgs) {
+            *dst++ = static_cast<uint32_t>(nb::cast<int32_t>(arg));
+        }
+    }
```

Comment:
do we need this specialization (between ScalarKind::F32 VS ScalarKind::I32)? Can we try treating everything as a uint32_t?

Reply:
IIRC, in push constants in IREE, i32 and f32 is treated as uint32_t*

Reply:
e.g
```cpp
for (iree_host_size_t i = 0; i < kernel_params->constant_count; i++) {
    *((uint32_t*)params_ptr[kernel_params->binding_count + i]) =
        ((const uint32_t*)constants.data)[i];
  }
```
https://github.com/iree-org/iree/blob/main/runtime/src/iree/hal/drivers/hip/graph_command_buffer.c#L800-L807

This way we can handle mixed of i32 and f32 inputs

Reply:
Yes, you are right. I tested it and it works.

Reply:
It is throwing `bad_cast` for float dtype.